### PR TITLE
Update the REST APIs to use the new response codes and auditing annotations.

### PR DESCRIPTION
**Pull request by mcpierce**
## Status
**HOLD**

I'm in the process of moving the APIs over, so don't merge this until I mark it as ready, please.

## Migrations
NO

## Description
Replaced the state wiring code. Updated the response type for the REST API.

### CODE REVIEW

File: comixed-frontend/src/app/backend-status/models/build-details.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export interface BuildDetails {
+export interface BuildDetail {
```
Comment (BRUCELLA2) [inline comixed-frontend/src/app/backend-status/models/build-details.ts]: the interface has been renamed, Shouldn't the file name be renamed accordingly?

**Comment by BRUCELLA2** [comment]
Just one question and there is some conflicting files : 
comixed-frontend/src/app/backend-status/backend-status.module.ts
comixed-frontend/src/app/backend-status/index.ts
comixed-rest-api/src/main/java/org/comixedproject/net/GetTaskAuditLogResponse.java

File: comixed-frontend/src/app/backend-status/models/build-details.ts (hunk 1 above)
Comment (mcpierce) [reply comixed-frontend/src/app/backend-status/models/build-details.ts]: I need to rename it back to BuildDetails, had meant to do it but forgot.

**Comment by mcpierce** [comment]
> Just one question and there is some conflicting files :
> comixed-frontend/src/app/backend-status/backend-status.module.ts
> comixed-frontend/src/app/backend-status/index.ts
> comixed-rest-api/src/main/java/org/comixedproject/net/GetTaskAuditLogResponse.java

Fixed.

**Comment by mcpierce** [comment]
Don't merge this one just yet, please.

**Pull request closed without merging**